api/lib/infrastructure/events/EventDispatcher.js

```diff
@@ -18,23 +18,50 @@ class EventDispatcher {
   }
 
   async dispatch(dispatchedEvent, domainTransaction) {
-    const subscriptions = this._subscriptions.filter(({ event }) => dispatchedEvent instanceof event);
 
-    for (const { eventHandler } of subscriptions) {
-      const returnedEventOrEvents = await eventHandler({ domainTransaction, event: dispatchedEvent });
-      await this._dispatchEventOrEvents(returnedEventOrEvents, domainTransaction);
+    const eventQueue = new EventQueue();
+    eventQueue.push(dispatchedEvent);
+
+    while (!eventQueue.isEmpty()) {
+      const eventToDispatch = eventQueue.shift();
+      const eventHandlers = this._findEventHandlersByEventType(eventToDispatch);
+
+      for (const eventHandler of eventHandlers) {
+        const resultingEventOrEvents = await eventHandler({ domainTransaction, event: eventToDispatch });
```

Comment:
On le fait dans une PR d'après de virer la domainTransaction du système d'event ?

Reply:
Faudra le faire oui, je sais plus si au final on l'a déjà viré de tous les handlers...

Reply:
Non not yet, à faire